opendevin/runtime/server/runtime.py

```diff
@@ -15,13 +16,25 @@
     NullObservation,
     Observation,
 )
+from opendevin.events.stream import EventStream
+from opendevin.runtime import Sandbox
 from opendevin.runtime.runtime import Runtime
+from opendevin.storage.local import LocalFileStore
 
 from .browse import browse
 from .files import read_file, write_file
 
 
 class ServerRuntime(Runtime):
+    def __init__(
+        self,
+        event_stream: EventStream,
+        sid: str = 'default',
+        sandbox: Sandbox | None = None,
+    ):
+        super().__init__(event_stream, sid, sandbox)
```

Comment:
It seems to be a problem here, when switching to a new agent, it will create a new runtime with the same event_stream, and the super init code: 
```
self.event_stream = event_stream
        self.event_stream.subscribe(EventStreamSubscriber.RUNTIME, self.on_event)
```
will throw exception like `ValueError: Subscriber already exists: runtime`

Reply:
Thanks for catching this! I was trying to figure out how @isavita triggered the "Subscriber already exists" error.

I think the right thing to do is start a new session every time settings change or the user resets. Otherwise we end up with a mixed event stream.

Going to refactor this now.

Reply:
I changed things so that you're forced into starting a new session if you want a new agent and/or runtime